### ISSUE TITLE
RAML nillable shorthand properties cannot have required facet

### ISSUE THREAD

**Issue by kdubb** [aml-org/amf#874]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [X] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

RAML nillable shorthand properties cannot have the `required` facet.

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

The following RAML fails to parse with error `Property 'required' not supported in a RAML 1.0 string node`.

```raml
types:
  Test:
    properties:
      name:
        required: false
        type: string?
```
Note that the expanded form (`type: (sting | nil)`) works correctly.

* **What is the expected behavior?**

Use of an explicit `required` facet should be allowed.

* **What is the motivation/use case for changing the behavior? (for feature requests)**

* **Please tell us about your environment:**
  
  - AMF Version: 4.7.1
  - AMF Distribution: JVM
  - Operating System: macOS
  - Link to your project: https://github.com/outfoxx/sunday-generator

* **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, Gitter, etc)

**Comment by AgustinBettati** [comment]
Thank you for creating the issue, internal reference: https://www.mulesoft.org/jira/browse/APIMF-2975

**Comment by arielmirra** [comment]
Hi @kdubb! We've fixed this issue!
Now the raml example provided doesn't throw an incorrect validation.

This fix will be present in the next AMF release approximately on August 23rd. It will also be available in AMF 5.0.0-beta.4 if you're considering switching to the new AMF major version.

Thanks!
